debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/pgoutput/PgOutputMessageDecoder.java

```diff
@@ -0,0 +1,635 @@
+
+        // Cache position as we're going to peak at the first byte to determine message type
+        // We need to reprocess all BEGIN/COMMIT messages regardless.
+        int position = buffer.position();
+        try {
+            MessageType type = MessageType.forType((char) buffer.get());
+            LOGGER.trace("Message Type: {}", type);
+            switch(type) {
+                case BEGIN:
+                case COMMIT:
+                case RELATION:
+                    // BEGIN/COMMIT
+                    // These types should always be processed due to the nature that they provide
+                    // the stream with pertinent per-transaction boundary state we will need to
+                    // always cache as we potentially might reprocess the stream from an earlier
+                    // LSN point.
+                    //
+                    // RELATION
+                    // These messages are always sent with a lastReceivedLSN=0; and we need to
+                    // always accept these to keep per-stream table state cached properly.
+                    LOGGER.trace("{} messages are always reprocessed", type);
+                    return false;
+                default:
+                    // INSERT/UPDATE/DELETE/TYPE/ORIGIN
+                    // These should be excluded based on the normal behavior, delegating to default method
+                    return MessageDecoder.super.shouldMessageBeSkipped(buffer, lastReceivedLsn, startLsn);
+            }
+        }
+        finally {
+            // Reset buffer position
+            buffer.position(position);
+        }
+    }
+
+    @Override
+    public void processMessage(ByteBuffer buffer, ReplicationMessageProcessor processor, TypeRegistry typeRegistry) throws SQLException, InterruptedException {
+        final MessageType messageType = MessageType.forType((char) buffer.get());
+        switch(messageType) {
+            case BEGIN:
+                handleBeginMessage(buffer);
+                break;
+            case COMMIT:
+                handleCommitMessage(buffer);
+                break;
+            case RELATION:
+                handleRelationMessage(buffer, typeRegistry);
+                break;
+            case INSERT:
+                decodeInsert(buffer, typeRegistry, processor);
+                break;
+            case UPDATE:
+                decodeUpdate(buffer, typeRegistry, processor);
+                break;
+            case DELETE:
+                decodeDelete(buffer, typeRegistry, processor);
+                break;
+            default:
+                LOGGER.trace("Message Type {} skipped, not processed.", messageType);
+                break;
+        }
+    }
+
+    @Override
+    public ChainedLogicalStreamBuilder optionsWithMetadata(ChainedLogicalStreamBuilder builder) {
+        return builder.withSlotOption("proto_version", 1)
+                .withSlotOption("publication_names", "dbz_publication");
```

Comment:
This name would have to be configurable, I reckon?

Reply:
Indeed, I added a `publication.name` configuration parameter yesterday so should be in next push.